packages/blockchain/egld/src/lib/services/egld.offchain.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiServices, Currency, TransferEth, TransferEthKMS, TransferEthMnemonic } from '@tatumio/api-client'
+import { abstractBlockchainOffchain } from '@tatumio/shared-blockchain-abstract'
+import { Blockchain } from '@tatumio/shared-core'
+import BigNumber from 'bignumber.js'
+import { from } from 'form-data'
+import { prepareSignedTransaction } from './egld.tx'
+
+export const egldOffchainService = (args: { blockchain: Blockchain }) => {
+  return {
+    ...abstractBlockchainOffchain(args),
+    send,
+  }
+}
+
+type EgldOffchain = TransferEth | TransferEthMnemonic | TransferEthKMS
+
+/**
+ * Send EGLD transaction from Tatum Ledger account to the blockchain. This method broadcasts signed transaction to the blockchain.
+ * This operation is irreversible.
+ * @param body content of the transaction to broadcast
+ * @returns transaction id of the transaction in the blockchain or id of the withdrawal, if it was not cancelled automatically
+ */
+export const send = async (body: EgldOffchain) => {
+  if ('signatureId' in body) {
+    return await ApiServices.offChain.blockchain.egldTransfer(body as TransferEthKMS)
+  }
+
+  const { gasLimit, gasPrice, ...withdrawal } = body
+  let fromPriv: string
+
+  if ('mnemonic' in body) {
```

Comment:
What about testing if there `privateKey` present in body.

Reply:
since I already checked if there is signatureId & mnemonic then there is only privateKey left so didn't think I had to check